Handle self disk

skycastlelily2 opened the pull request:
Hi Cole,
   Happy New year! , and here are the commits^^

crobinso commented:
Hi, sorry for the delay, I only returned to work this week.

`./setup.py pylint` flags some warnings, please make sure to run that on every patch to make sure things are clean, and ensure pylint and pycodestyle are installed to get full coverage. You have some trailing whitespace issues, there's probably a way to get your editor to flag those issues for you

```
$ ./setup.py pylint
running pylint
running codespell
running pycodestyle
virtManager/delete.py:483: [W293] blank line contains whitespace
virtManager/delete.py:511: [W291] trailing whitespace
virtManager/delete.py:521: [W293] blank line contains whitespace
virtManager/delete.py:525: [W293] blank line contains whitespace
virtManager/delete.py:540: [W291] trailing whitespace
virtManager/delete.py:554: [W291] trailing whitespace
running pylint
************* Module virtManager.delete
virtManager/delete.py:483:0: C0303: Trailing whitespace (trailing-whitespace)
virtManager/delete.py:511:18: C0303: Trailing whitespace (trailing-whitespace)
virtManager/delete.py:521:0: C0303: Trailing whitespace (trailing-whitespace)
virtManager/delete.py:525:0: C0303: Trailing whitespace (trailing-whitespace)
virtManager/delete.py:540:28: C0303: Trailing whitespace (trailing-whitespace)
virtManager/delete.py:554:18: C0303: Trailing whitespace (trailing-whitespace)
virtManager/delete.py:518:16: E0602: Undefined variable 'storage_errors' (undefined-variable)
virtManager/delete.py:561:16: E0602: Undefined variable 'storage_errors' (undefined-variable)
```

* Rename `_remove_the_devices` to `_delete_disks`, it's more clear
* The _delete_vm implementation for vmmDeleteStorage should not do anything, remember we don't want to remove the VM when just removing a single disk. It should just be 'pass'
* Please rebase to master, and incorporate the changes from commit b1b9e4b6887

crobinso commented on virtManager/delete.py:
```diff
@@ -493,3 +491,25 @@ def append_str(str1, str2, delim="\n"):
+
+
+class vmmDeleteDialog(_vmmDeleteBase):
+
```
Please move these classes up to be right after the _vmmDeleteBase class, before the 'UI init helpers' comment block
Also remove the extra newline here, before `__init__(`

crobinso commented on virtManager/delete.py:
```diff
@@ -493,3 +491,25 @@ def append_str(str1, str2, delim="\n"):
+
+    def __init__(self):
+        _vmmDeleteBase.__init__(self)
+
```
This __init__function doesn't do anything, you can remove it.

crobinso commented on virtManager/delete.py:
```diff
@@ -493,3 +491,25 @@ def append_str(str1, str2, delim="\n"):
+    disk = None
+    def __init__(self):
+        _vmmDeleteBase.__init__(self)
+
```
I think you can change the __init__ function here to be

```python
def __init__(self, disk):
    _vmmDeleteBase.__init__(self)
    self.disk = disk
```

Rather than have a separate `set_disk` function

crobinso commented on virtManager/delete.py:
```diff
@@ -504,17 +495,35 @@ def _get_dialog_title(self):
         _vmmDeleteBase.__init__(self)
 
+
     def _get_dialog_title(self):
```
Random space addition here

crobinso commented on virtManager/delete.py:
```diff
@@ -504,17 +495,35 @@ def _get_dialog_title(self):
+        text = _('%s') % self.disk.target
+        return [title, text]
+
+
```
Extra newline addition here, just use single line between them for consistency

crobinso commented on virtManager/delete.py:
```diff
@@ -505,6 +508,11 @@ def _get_title_text(self, devs):
+        vm_active = self.vm.is_active()
+        return vm_active
+ 
+
```
Stick with two newlines between the class definitions, after this it is 3

crobinso commented on virtManager/delete.py:
```diff
@@ -524,6 +532,9 @@ def _get_title_text(self, devs):
         text = _('%s') % self.disk.target
         return [title, text]
 
+    def _vm_active_status(self)
```
This should always return False. We do not want to show a warning about the active VM when just removing storage.

crobinso commented on virtManager/delete.py:
```diff
@@ -508,10 +508,14 @@ def _get_title_text(self, devs):
         return [title, text]
 
-    def _vm_active_status(self)
+    def _vm_active_status(self):
```
whoops, this bit needs to be squashed into the previous patch. Each individual commit should be working and pass pylint. `git rebase --interactive` is your friend :)

crobinso commented on virtManager/delete.py:
```diff
@@ -508,10 +508,14 @@ def _get_title_text(self, devs):
         return vm_active
- 
+
+    def _delete_vm(self, vm, undefine):
```
I don't think this function needs to take the 'undefine' value. In the caller, `undefine = vm.is_transient()` which you can check here easily enough.

crobinso commented on virtManager/delete.py:
```diff
@@ -517,6 +507,21 @@ def _delete_vm(self, vm, undefine):
+    def _delete_disks(self, vm, paths, conn, meter):
+        storage_errors = []
+        if not paths:
+            return
```
This `if not paths` conditional can be removed, it's left over from the previous mixed storage vs vm implementation.

crobinso commented on virtManager/delete.py:
```diff
@@ -209,26 +211,14 @@ def _async_delete(self, asyncjob, vm, paths):
                       (vm.get_name(), str(e)))
             details = "".join(traceback.format_exc())
 
-
```
More whitespace changes.

crobinso commented on virtManager/delete.py:
```diff
@@ -543,5 +548,21 @@ def _vm_active_status(self):
+                storage_errors.append((str(e),
+                                          "".join(traceback.format_exc())))
+            meter.end(0)
+        return storage_errors
```
This block is now duplicated twice. It should be shared. Also vm.remove_device is only called when no paths are selected for deletion? In this case it looks like we should call it unconditionally, right?

skycastlelily2 commented:
Hi Cole,
   Thanks a lot for your time on helping me polish the commits,I do feel have learned a lot.I have git rebase master -i again ,modified the code accordingly and added a _destroy_vm function:)

crobinso commented:
I tweaked the patches and commit messages a bit, and pushed now!
